halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/providers/kubernetes/KubernetesEditAccountCommand.java

```diff
@@ -0,0 +1,129 @@
+import com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesAccount;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class KubernetesEditAccountCommand extends AbstractEditAccountCommand<KubernetesAccount> {
+  protected String getProviderName() {
+    return "kubernetes";
+  }
+
+  @Parameter(
+      names = "--context",
+      description = "The kubernetes context to be managed by Spinnaker. "
+          + "See http://kubernetes.io/docs/user-guide/kubeconfig-file/#context for more information."
+  )
+  private String context;
+
+  @Parameter(
+      names = "--clear-context",
+      description = "Removes the currently specified context, defaulting to 'current-context' in your kubeconfig."
+          + "See http://kubernetes.io/docs/user-guide/kubeconfig-file/#context for more information."
+  )
+  private boolean clearContext;
+
+  @Parameter(
+      names = "--namespaces",
+      variableArity = true,
+      description = "A list of namespaces this Spinnaker account can deploy to and will cache."
+  )
+  private List<String> namespaces = new ArrayList<>();
+
+  @Parameter(
+      names = "--all-namespaces",
+      description = "Set the list of namespaces to cache and deploy to every namespace available to your supplied credentials."
+  )
+  private boolean allNamespaces;
+
+  @Parameter(
+      names = "--add-namespace",
+      description = "Add this namespace to the list of namespaces to manage."
+  )
+  private String addNamespace;
+
+  @Parameter(
+      names = "--remove-namespace",
+      description = "Remove this namespace to the list of namespaces to manage."
+  )
+  private String removeNamespace;
+
+  @Parameter(
+      names = "--docker-registries",
+      variableArity = true,
+      description = "A list of the Spinnaker docker registry account names this Spinnaker account can use as image sources. "
+          + "These docker registry accounts must be registered in your halconfig before you can add them here."
+  )
+  public List<String> dockerRegistries = new ArrayList<>();
+
+  @Parameter(
+      names = "--add-docker-registry",
+      description = "Add this docker registry to the list of docker registries to use as a source of images."
+  )
+  private String addDockerRegistry;
+
+  @Parameter(
+      names = "--remove-docker-registry",
+      description = "Remove this docker registry from the list of docker registries to use as a source of images."
+  )
+  private String removeDockerRegistry;
+
+  @Override
+  protected Account editAccount(KubernetesAccount account) {
+    boolean contextSet = context != null && !context.isEmpty();
+    if (contextSet && !clearContext) {
+      account.setContext(context);
+    } else if (!contextSet && clearContext) {
+      account.setContext(null);
+    } else if (contextSet && clearContext) {
+      throw new IllegalArgumentException("Both --context and --clear-context cannot be set");
```

Comment:
Feel free to ignore this comment entirely, but I think it is sometimes clearer to say exactly what to do instead of what not to do. So something like "Specify either --context or --clear-context." Eh, maybe not. I read it too many times and can't tell anymore...